README.md

```diff
@@ -60,6 +61,35 @@ which contain a toctree need to have the suffix `.trst` instead of just `.rst`.
 This way the projects which use this as a submodule are able to ignore the
 toctrees and build their own instead.
 
+### Understanding Spell Check
+A spell check is performed during the build step by default. It uses in-built
+language specific dictionaries and project specific dictionaries (aspell.en.pws)
+to verify the spellings and causes the build to fail in case of any typos.
+
+The project, which uses this blueprint, should have its own custom dictionary,
+similar to the one in blueprint (aspell.en.pws). A link to the check_spelling.sh
+in the repository can be created and the spell checker should then run correctly.
+
+To build without spell check, configure cmake as below:
+
+    cmake -H. -Bbuild -DPERFORM_SPELL_CHECK=OFF
+
+To invoke the spell check utility directly, run the command below:
+
+    ./check_spelling.sh
+
+The spell checker has various modes and options, which can be seen by:
+
+    ./check_spelling.sh -usage
```

Comment:
Can we have double dashes for `--usage` ? Also, the comvention is to call such a flag `--help` - but I am OK w/ usage.

If you want a single dash altenrative, I suggest `-u` or `-h`.

Reply:
Guess we should support both short and long options.